src/span.js

```diff
@@ -180,7 +204,7 @@ export default class Span {
             delete keyValuePairs[opentracing_tags.SAMPLING_PRIORITY];
         }
 
-        if (this._spanContext.isSampled()) {
+        if (this.isSampledAndFinished()) {
```

Comment:
1. I suggest not using the word "finished" because it clashes with finishing the span. I suggest "final" as in "sampling decision is final / finalized'.
1. assuming "finished" here means the same as "final", then the logic is wrong. The right way would be "sampled OR finalized" (with a better naming).
1. for the purpose of these update functions (set()/log()), it is better to abstract away the question of sampling. For example, if you have a method `isWriteable()`, it becomes very single-purpose and obvious how to use here. The actual implementation of the method may be checking several conditions without having a weird method name. The span is writeable is if is either (1) sampled for sure (final), (2) maybe sampled, (3) not finish()'d.